External Sources/PrologDoc/prologdoc-1.0RC2/pldoc.1
.\" Automatically generated by Pod::Man v1.37, Pod::Parser v1.14
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sh \" Subsection heading
.br
.if t .Sp
.ne 5
.PP
\fB\\$1\fR
.PP
..
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  | will give a
.\" real vertical bar.  \*(C+ will give a nicer C++.  Capital omega is used to
.\" do unbreakable dashes and therefore won't be available.  \*(C` and \*(C'
.\" expand to `' in nroff, nothing in troff, for use with C<>.
.tr \(*W-|\(bv\*(Tr
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
'br\}
.\"
.\" If the F register is turned on, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.Sh), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.if \nF \{\
.    de IX
.    tm Index:\\$1\t\\n%\t"\\$2"
..
.    nr % 0
.    rr F
.\}
.\"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.hy 0
.if n .na
.\"
.\" Accent mark definitions (@(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2).
.\" Fear.  Run.  Save yourself.  No user-serviceable parts.
.    \" fudge factors for nroff and troff
.if n \{\
.    ds #H 0
.    ds #V .8m
.    ds #F .3m
.    ds #[ \f1
.    ds #] \fP
.\}
.if t \{\
.    ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.    ds #V .6m
.    ds #F 0
.    ds #[ \&
.    ds #] \&
.\}
.    \" simple accents for nroff and troff
.if n \{\
.    ds ' \&
.    ds ` \&
.    ds ^ \&
.    ds , \&
.    ds ~ ~
.    ds /
.\}
.if t \{\
.    ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.    ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.    ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.    ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.    ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.    ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.\}
.    \" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.    \" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.    \" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.    ds : e
.    ds 8 ss
.    ds o a
.    ds d- d\h'-1'\(ga
.    ds D- D\h'-1'\(hy
.    ds th \o'bp'
.    ds Th \o'LP'
.    ds ae ae
.    ds Ae AE
.\}
.rm #[ #] #H #V #F C
.\" ========================================================================
.\"
.IX Title "PLDOC.IN 1"
.TH PLDOC.IN 1 "2004-11-12" "PrologDoc 1.0RC2" "Prolog Utilities"
.SH "NAME"
pldoc \- generate documentation from prolog source files
.SH "USAGE"
.IX Header "USAGE"
.Vb 5
\&  $ pldoc [-v|--verbose] [-l|--load preload_file] \e
\&          [-p|--prefix prefix_for_html_elements] \e
\&          [-s|--side-effects] [-f|--footer <footer>] \e
\&          [-t|--to dest_dir] \e
\&          file1.pl file2.pl ...
.Ve
.Sh "Options"
.IX Subsection "Options"
The options accepted by pldoc are:
.IP "\-\-verbose" 4
.IX Item "--verbose"
.PD 0
.IP "\-v" 4
.IX Item "-v"
.PD
verbose mode
.IP "\-\-prefix <prefix_for_html_elements>" 4
.IX Item "--prefix <prefix_for_html_elements>"
.PD 0
.IP "\-p <prefix_for_html_elements>" 4
.IX Item "-p <prefix_for_html_elements>"
.PD
prefix references to external \s-1HTML\s0 elements as \s-1CSS\s0 style sheets and
images with the passed argument. This is useful if you want to publish
the generated documentation on a web server, i.e.:
.Sp
.Vb 1
\&  pldoc --prefix http://foo-prolog.com/docs/style_sheets/ foo.pl
.Ve
.Sp
The default prefix is the absolute path of the PrologDoc
library, i.e. \f(CW\*(C`/usr/local/lib/prologdoc/css/\*(C'\fR.
.IP "\-\-load <prolog_file>" 4
.IX Item "--load <prolog_file>"
.PD 0
.IP "\-l <prolog_file>" 4
.IX Item "-l <prolog_file>"
.PD
prolog file containing operator definitions to preload before parsing the
files to document.
.IP "\-\-side\-effects" 4
.IX Item "--side-effects"
.PD 0
.IP "\-s" 4
.IX Item "-s"
.PD
document side effects.
.IP "\-\-footer <footer>" 4
.IX Item "--footer <footer>"
.PD 0
.IP "\-f <footer>" 4
.IX Item "-f <footer>"
.PD
footer to include on \s-1HTML\s0 generated pages, i.e.:
.Sp
.Vb 1
\&  pldoc -f "Copyright (c) 2004 Homer Simpson. All rights reserved." peanuts.pl
.Ve
.IP "\-\-to <destination_dir>" 4
.IX Item "--to <destination_dir>"
.PD 0
.IP "\-t <destination_dir>" 4
.IX Item "-t <destination_dir>"
.PD
destination directory to place the generated documents
.SH "COPYRIGHT AND LICENSE"
.IX Header "COPYRIGHT AND LICENSE"
Copyright (C) 1999 by Elisheva Bonchek (le7bonch \s-1AT\s0 cs.huji.ac.il)
and Sara Cohen (sarac \s-1AT\s0 ie.technion.ac.il, http://iew3.technion.ac.il/~sarac).
Copyright (C) 2004 by Bram Adams (bram.adams \s-1AT\s0 ugent.be).
Copyright (C) 2004 by Salvador Fandino (sfandino@yahoo.com).
.PP
PrologDoc is free software; you can redistribute it and/or modify it
under the terms of the \s-1GNU\s0 General Public License as published by the
Free Software Foundation; either version 2 of the License, or (at your
option) any later version.
.PP
PrologDoc is distributed in the hope that it will be useful, but
\&\s-1WITHOUT\s0 \s-1ANY\s0 \s-1WARRANTY\s0; without even the implied warranty of
\&\s-1MERCHANTABILITY\s0 or \s-1FITNESS\s0 \s-1FOR\s0 A \s-1PARTICULAR\s0 \s-1PURPOSE\s0.  See the \s-1GNU\s0
General Public License for more details.
.PP
You should have received a copy of the \s-1GNU\s0 General Public License
along with PrologDoc; if not, write to the Free Software Foundation,
Inc., 59 Temple Place, Suite 330, Boston, \s-1MA\s0 02111\-1307 \s-1USA\s0


External Sources/PrologDoc/prologdoc-1.0RC2/pldoc.pl
/* -*- Mode: Prolog -*-

  This file is part of PrologDoc (http://prologdoc.sourceforge.net/).

  Copyright (C) 1999 by Elisheva Bonchek (le7bonch AT cs.huji.ac.il) and Sara Cohen (sarac AT ie.technion.ac.il, http://iew3.technion.ac.il/~sarac) 
  Copyright (C) 2004 by Bram Adams (bram.adams AT ugent.be)
  Copyright (C) 2004 by Salvador Fandino (sfandino@yahoo.com)

  PrologDoc is free software; you can redistribute it and/or modify
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation; either version 2 of the License, or
  (at your option) any later version.

  PrologDoc is distributed in the hope that it will be useful,
  but WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  GNU General Public License for more details.

  You should have received a copy of the GNU General Public License
  along with PrologDoc; if not, write to the Free Software
  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/

:- consult(prolog_doc(prolog_doc)).

main :-
	style_check(-atom),  %kill string length limit
	current_prolog_flag(argv, AllArgs),
	(   append(_, ['--', Prefix, SideEffects,
		       Footer, Load, To|Files], AllArgs)
	->  setConfig(html_prefix, Prefix),
	    setConfig(include_side_effects, SideEffects),
	    setConfig(html_footer, Footer),
	    prologDoc(To, Files, Load)
	;   writef("invalid arguments: %q", AllArgs) ).
	
